kousa/lib/gen/voice_rabbit.ex

```diff
@@ -25,19 +25,26 @@ defmodule Kousa.Gen.VoiceRabbit do
   @send_queue "shawarma_queue"
   @receive_exchange "kousa_exchange"
   @receive_queue "kousa_queue"
+  @retry_interval 15_000
 
   def init(opts) do
-    {:ok, conn} =
-      Connection.open(Application.get_env(:kousa, :rabbit_url, "amqp://guest:guest@localhost"))
-
-    {:ok, chan} = Channel.open(conn)
-    setup_queue(opts.id, chan)
-
-    queue_to_consume = @receive_queue <> opts.id
-    IO.puts("queue_to_consume: " <> queue_to_consume)
-    # Register the GenServer process as a consumer
-    {:ok, _consumer_tag} = Basic.consume(chan, queue_to_consume, nil, no_ack: true)
-    {:ok, %State{chan: chan, id: opts.id}}
+    host = Application.get_env(:kousa, :rabbit_url, "amqp://guest:guest@localhost")
+
+    case Connection.open(host) do
+      {:ok, conn} ->
+        {:ok, chan} = Channel.open(conn)
+        setup_queue(opts.id, chan)
+
+        queue_to_consume = @receive_queue <> opts.id
+        IO.puts("queue_to_consume: " <> queue_to_consume)
+        # Register the GenServer process as a consumer
+        {:ok, _consumer_tag} = Basic.consume(chan, queue_to_consume, nil, no_ack: true)
+        {:ok, %State{chan: chan, id: opts.id}}
+      {:error, _} ->
+        IO.puts("[Rabbit] error on connecting to server: #{host}")
+        :timer.sleep(@retry_interval)
+        init(opts)
+    end
```

Comment:
does this only work on initial startup? What about when the connection gets dropped after init?

Reply:
Yes. I should have mentioned that this is only for initial startup to address an issue I was having with syncing containers. I can extend this to address dropped connections as well.

Reply:
I'm not familiar with Elixir well enough to work with the supervisor pattern, but I believe that's what would need to be used to restart the RabbitMQ-related processes.  Prior to the changes in this PR, the application completely crashes if no RabbitMQ connection is established.  After this PR, the application completely crashes if no RabbitMQ connection established except for during initial startup.  Can the reconnect logic be added in another PR for the time being? I did add retry logic to Shawarma though that handles disconnects even after init.

Reply:
I was able to update the logic to enable reconnecting with amqp after all.